lib/improver/tests/metadata/test_utilities.py

```diff
@@ -31,12 +31,63 @@
 """Tests for the improver.metadata.utilities module"""
 
 import unittest
+import iris
 import numpy as np
 
-from improver.metadata.utilities import generate_hash, create_coordinate_hash
+from improver.metadata.utilities import (
+    create_new_diagnostic_cube, generate_hash, create_coordinate_hash)
 from improver.tests.set_up_test_cubes import set_up_variable_cube
 
 
+class Test_create_new_diagnostic_cube(unittest.TestCase):
+    """Test utility to create new diagnostic cubes"""
+
+    def setUp(self):
+        """Set up template with data, coordinates, attributes and cell
+        methods"""
+        self.template_cube = set_up_variable_cube(
+            280*np.ones((3, 5, 5), dtype=np.float32),
+            standard_grid_metadata='uk_det')
+        self.template_cube.add_cell_method('time (max): 1 hour')
+        self.name = "lwe_precipitation_rate"
+        self.units = "mm h-1"
+
+    def test_basic(self):
+        """Test result is a cube that inherits coordinates only"""
+        result = create_new_diagnostic_cube(
+            self.name, self.units, self.template_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.standard_name, "lwe_precipitation_rate")
+        self.assertEqual(result.units, "mm h-1")
+        self.assertSequenceEqual(result.coords(), self.template_cube.coords())
+        self.assertSequenceEqual(result.coords(dim_coords=True),
+                                 self.template_cube.coords(dim_coords=True))
```

Comment:
I think this is already implicitly tested on the line above.

Reply:
I wasn't sure - since I found `assert` didn't check things like datatypes, I wasn't sure what other properties wouldn't be checked here.

Reply:
I've split these up explicitly.